src/test/java/io/strimzi/kafka/bridge/tracing/TracingTestBase.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.kafka.bridge.tracing;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.strimzi.kafka.bridge.BridgeContentType;
+import io.strimzi.kafka.bridge.http.services.ProducerService;
+import io.strimzi.kafka.bridge.utils.Urls;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.tracing.TracingOptions;
+import io.vertx.core.tracing.TracingPolicy;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URL;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * Base for OpenTracing and OpenTelemetry (manual) tests.
+ * <p>
+ * Test will only run if the bridge AND tracing server are up-n-running.
+ */
+@ExtendWith(VertxExtension.class)
+public abstract class TracingTestBase {
+    Logger log = LoggerFactory.getLogger(getClass());
+
+    private void assumeServer(String url) {
+        try {
+            new URL(url).openConnection().getInputStream();
+        } catch (Exception e) {
+            log.info("Cannot connect to server", e);
+            Assumptions.assumeTrue(false, "Server is not running: " + url);
+        }
+    }
+
+    Handler<AsyncResult<HttpResponse<JsonObject>>> verifyOK(VertxTestContext context) {
+        return ar -> {
+            context.verify(() -> {
+                assertThat(ar.succeeded(), is(true));
+                HttpResponse<JsonObject> response = ar.result();
+                assertThat(response.statusCode(), is(HttpResponseStatus.OK.code()));
+            });
+            context.completeNow();
+        };
+    }
+
+    @BeforeEach
+    public void setUp() {
+        assumeServer(String.format("http://%s:%s", Urls.BRIDGE_HOST, Urls.BRIDGE_PORT)); // bridge
+        assumeServer("http://localhost:16686"); // jaeger
```

Comment:
I think it would be better if we could use testcontainers to spin up a server. I admit I've no idea how much work that would really, but maybe it's not so much?

Reply:
With testcontainers it's chicken-n-egg problem ... you would first need to build latest Bridge image, only to be used in tests then.
Perhaps I can easily make it work with running Application::main with the right args ...

Reply:
Ah, now I know why I added that service.name property to Bridge config ... otherwise you can only enable OpenTracing via env vars ... which makes it hard (if not impossible) to test.

Reply:
Something like this is then required - custom generic container:
* https://github.com/alesj/test-clients/tree/tests1/testsuite/src/test/java/io/strimzi/testclients/test/support

I can give it a shot if it's really needed ...

Reply:
Looking at Ales's branch I'm inclined to thiat bridge support for the test-clients would be a great addition. @ppatierno @scholzj wdyt?